apps/pkcs12.c

```diff
@@ -669,9 +677,30 @@ int pkcs12_main(int argc, char **argv)
         if (!twopass)
             OPENSSL_strlcpy(macpass, pass, sizeof(macpass));
 
-        p12 = PKCS12_create_ex(cpass, name, key, ee_cert, certs,
-                               key_pbe, cert_pbe, iter, -1, keytype,
-                               app_get0_libctx(), app_get0_propq());
+        /* Load the config file */
+        if ((conf = app_load_config(default_config_file)) == NULL)
+            goto export_end;
+        if (!app_load_modules(conf))
+            goto export_end;
+        /* Find the cert bag section */
+        if ((cb_attr = NCONF_get_string(conf, "pkcs12", "certBagAttr")) != NULL) {
+            if ((cb_sk = NCONF_get_section(conf, cb_attr)) != NULL) {
+                if (sk_CONF_VALUE_num(cb_sk) > 0) {
+                    val = sk_CONF_VALUE_value(cb_sk, i);
+                    if (strcmp(val->name, "jdkTrustedKeyUsage") == 0)
+                        obj = OBJ_txt2obj(val->value, 0);
+                }
+            } else {
+                ERR_clear_error();
+            }
+        } else {
+            ERR_clear_error();
+        }
```

Comment:
All of this does make me wonder whether the config file is strictly necessary. Could we do all of this just on the basis of a command line arg? Or possibly both command line arg and config file (users can use whichever they prefer).

Reply:
this was the original idea I believe, a new CLI argument but the convo moved to it becoming a config option

Reply:
@mattcaswell 

> All of this does make me wonder whether the config file is strictly necessary. Could we do all of this just on the basis of a command line arg? Or possibly both command line arg and config file (users can use whichever they prefer).

Config file was my idea because that avoids hard-coding single purpose command line switches (which was the original idea).

Both config and command line would be nice, but command line options should then follow the existing syntax for adding extensions, not be a single purpose switch (because I am sure you wouldn't want proliferation of vendor specific switches for everything).